### PR TITLE
Adding tabs guidelines

### PR DESCRIPTION
**Pull request by gdoyle1**
This PR adds tabs design guidelines

### CODE REVIEW

File: packages/v4/src/content/design-guidelines/components/tabs/tabs.md [hunk 1]
```diff
@@ -2,4 +2,64 @@
...
+
+## Usage
+
+PatternFly offers a few types of tabs:
+
+- [Default tabs](#default-tabs) (recommended)
+- [Boxed tabs](boxed-tabs)
+- [Vertical tabs](#vertical-tabs)
+
+You can use as many tabs as you want in a page, although it is recommended you keep the total number of tabs to a minimum for a better user experience.If you have more tabs than fit on the page, you may use a [tab overflow](/components/tabs#default-overflow) which would allow a user to horizontal scroll through your tabs. The uses of both default and boxed stylings are flexible, so use either according to your needs and preferences.
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/tabs/tabs.md:15]: Missing a space after the first sentence.

Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/tabs/tabs.md:15]: Tab overflow is called "Default overflow" on the React page.

File: packages/v4/src/content/design-guidelines/components/tabs/tabs.md [hunk 2]
```diff
@@ -2,4 +2,64 @@
...
+
+You can use as many tabs as you want in a page, although it is recommended you keep the total number of tabs to a minimum for a better user experience.If you have more tabs than fit on the page, you may use a [tab overflow](/components/tabs#default-overflow) which would allow a user to horizontal scroll through your tabs. The uses of both default and boxed stylings are flexible, so use either according to your needs and preferences. 
+
+### Default tabs
+
+Default tabs do not have any borders, other than the bottom line to distinguish between a selected tab, a hovered over tab, and a non-active tab. These tabs are most commonly used as top page header tabs or nested within components.
+
+<img src="./img/default-tabs.png" alt="Default tabs ex" width="1500"/>
+
+Default tabs can be formatted as standard or filled.  Filled tabs are placed side by side and span the whole width of the page. They can be used in screens with narrower widths, such as a drawer within a [primary detail view](/demos/primary-detail). When in doubt, stick with the standard styled tabs.
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/tabs/tabs.md:23]: The use of "Filled styled" was a little confusing. The tabs themselves are not filled. It's more of a question of whether the tabs should stretch to fit to the container width or not. Maybe "standard" or "fit to container" would be clearer?

File: packages/v4/src/content/design-guidelines/components/tabs/tabs.md [hunk 3]
```diff
@@ -2,4 +2,64 @@
...
+### Secondary tabs
+
+Use these for any use case that differs from primary tabs. Secondary tabs use 14px text. They are especially useful when another set of tabs are needed after primary as well as when tabs are needed within a component.
+
+<img src="./img/secondary-1-tabs.png" alt="Secondary tabs ex" width="1500"/>
+
+1. Primary tabs
+2. Secondary tabs
+
+<img src="./img/secondary-2-tabs.png" alt="Secondary tabs ex" width="1500"/>
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/tabs/tabs.md:65]: Are you going to say anything about the last example? Seems to be just hanging there.

**Review by mcarrano — CHANGES_REQUESTED** [review]
This looks good overall @gdoyle1 . Just a few minor issues.

File: packages/v4/src/content/design-guidelines/components/tabs/tabs.md [hunk 4]
```diff
@@ -2,4 +2,64 @@
...
+
+### Vertical tabs
+
+Vertical tabs are placed on the left hand side of a page or container. 
+
+<img src="./img/vertical-tabs.png" alt="Vertical tabs ex" width="1500"/>
+
+## Hierarchy
+
+Both boxed and default tabs can help create hierarchy on the page. We recommend using default tabs at all times, although mixing the two styles can separate sections of the page more distinctly. When using many levels of default tabs, make sure to incorporate the following types.
```
Comment (mmenestr) [inline packages/v4/src/content/design-guidelines/components/tabs/tabs.md:44]: Re-reading this, I feel like it's a little bit confusing because there is no clear distinction between what a primary tab is as opposed to a boxed tab or default tab?

I maybe would re-word to say, "When you have more than one tab level in your UI, you can either chose to use a mixture of boxed and default tabs to create hierarchy on a page, or you can chose to use primary and secondary tabs which differ in their font size, with primary tabs using 16pt font, and secondary tabs using 14pt font."

Or something like that? It's definitely hard to put into words/explain though, but do you see what I mean?

Maybe i'm misunderstanding the whole point of this section though. Are you just talking about information hierarchy in general, and not necessarily a "2 tab level" situation, or are you only talking about 2 tab level situations? or both?

File: packages/v4/src/content/design-guidelines/components/tabs/tabs.md (hunk 3 above)
Comment (mmenestr) [reply packages/v4/src/content/design-guidelines/components/tabs/tabs.md:65]: Agreed^ would maybe move the image up, so that both images are together, and change the little call out to have a "2" in it to show that it's a secondary tab as well.

Although now i'm confused about primary and secondary tabs. I thought secondary tabs should only be used when there are 2 levels of tabs, and in this modal example there's only one level so why would it be a secondary tab?

File: packages/v4/src/content/design-guidelines/components/tabs/tabs.md [hunk 5]
```diff
@@ -2,4 +2,64 @@
...
+
+### Primary tabs
+
+Use these only for top page header tabs and vertical tabs. Primary tabs use 16px text. 
+
+<img src="./img/primary-tabs.png" alt="Primary tabs ex" width="1500"/>
+
+### Secondary tabs
+
+Use these for any use case that differs from primary tabs. Secondary tabs use 14px text. They are especially useful when another set of tabs are needed after primary as well as when tabs are needed within a component.
```
Comment (mmenestr) [inline packages/v4/src/content/design-guidelines/components/tabs/tabs.md:58]: Would also just reword this to say "Use secondary tabs when, you have another set of tabs, in addition to a primary level, or when using tabs within a component, for example in a modal. Secondary tabs use 14px text."

File: packages/v4/src/content/design-guidelines/components/tabs/tabs.md (hunk 2 above)
Comment (gdoyle1) [reply packages/v4/src/content/design-guidelines/components/tabs/tabs.md:23]: The reason I used "filled" was because that's what it is called in the code https://www.patternfly.org/v4/components/tabs#filled. I can update the description here to be clearer though!

File: packages/v4/src/content/design-guidelines/components/tabs/tabs.md (hunk 4 above)
Comment (gdoyle1) [reply packages/v4/src/content/design-guidelines/components/tabs/tabs.md]: Added some clarification to why you'd use one over the other - let me know what you think @mmenestr

File: packages/v4/src/content/design-guidelines/components/tabs/tabs.md (hunk 3 above)
Comment (gdoyle1) [reply packages/v4/src/content/design-guidelines/components/tabs/tabs.md]: Yes added some text! @mcarrano @mmenestr

**Review by mcarrano — APPROVED** [review]
Thanks @gdoyle1 . Looks good to me!

**Review by mmenestr — APPROVED** [review]

File: packages/v4/src/content/design-guidelines/components/tabs/tabs.md (hunk 4 above)
Comment (mmenestr) [reply packages/v4/src/content/design-guidelines/components/tabs/tabs.md]: Looks good, thank you!!
